### ISSUE TITLE
Feature Request - WP Network integration

### ISSUE THREAD

**Issue by SkyeMoroney** [pluginkollektiv/blacklist-updater#7]
Hi Awesome people,

We use WP Networks to manage a number of school sites and this would be great. I tried installing it on a network to test but it does not currently appear to support wp-networks.

Keep on making the world better.

**Comment by swissspidy** [comment]
With WP Networks you just mean WordPress Multisite?

The plugin uses `update_site_option`, which means it updates the option network-wide if used on a Multisite install.

Can you elaborate on what doesn't work exactly?

**Comment by SkyeMoroney** [comment]
=( I never saw the update email and only just came across your response.
Yes, I did mean WordPress Multisite.

I no longer work at that site and I can't confirm what part didn't work anymore. Lets chalk it up to a misunderstanding of how to use it in a Multisite environment.

**Comment by utzer** [comment]
@swissspidy it seems the blacklist is only updated in the main page. I have a multi site (Wordpress Network) and after activating the blacklist field in the main site (the initial site before activating the network option) is filled with the blacklist words.

**Comment by swissspidy** [comment]
I‘m not really working on this anymore. I suggest opening a new issue.

**Comment by utzer** [comment]
@swissspidy ah sorry, I confused and thought you're the owner of the repo, but that would be @Zodiac1978 then.

**Comment by swissspidy** [comment]
The owner is the pluginkollektiv organization, not a single individual.

**Comment by scrobbleme** [comment]
This issue should be opened again. 
For us, it also only updates the list on the main site and not on other sites in the network...

**Comment by Zodiac1978** [comment]
> The plugin uses update_site_option, which means it updates the option network-wide if used on a Multisite install.

Yes, but only our option to save the etag. The disallowed keys are updated with `update_option` and therefore only affect the main site in a multisite context:
https://github.com/pluginkollektiv/blacklist-updater/blob/master/inc/class-blacklist-updater.php#L137-L143

A solution could be to loop through every site (simplified code example):
```php
function update_disallowed_keys_on_all_sites( $keys ) {
    $sites = get_sites();

    foreach ( $sites as $site ) {
        switch_to_blog( $site->blog_id );
        update_option( 'disallowed_keys', $keys );
        restore_current_blog();
    }
}
```

But this would change the behavior of the plugin, and at least should be made very clear in the update/changelog.

On the other side, I wouldn't recommend adding so many (similar!) keys to every site and bloat the database unnecessarily.

As we are overriding the complete option with the new file from GitHub (see #12) we also could save the keys in a new network option and append it to the local ones. This would solve #12, but is also a change in behavior. And the keys are not viewable per default (but we could add this to the network settings).

What do you all think? (If you are still interested in this after 10 years, for some ...)
